pkgs/tools/package-management/apt/default.nix

```diff
@@ -27,15 +26,15 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ pkgconfig ];
 
   buildInputs = [
-    cmake perl curl gtest lzma bzip2 lz4 db dpkg libxslt.bin
+    cmake perlPackages.perl curl gtest lzma bzip2 lz4 db dpkg libxslt.bin
   ] ++ lib.optionals withDocs [
-    doxygen Po4a w3m docbook_xml_dtd_45
+    doxygen perlPackages.Po4a w3m docbook_xml_dtd_45
   ] ++ lib.optionals withNLS [
     gettext
   ];
 
   preConfigure = ''
-    export PERL5LIB="$PERL5LIB''${PERL5LIB:+:}${Po4a}/lib/perl5";
+    export PERL5LIB="$PERL5LIB''${PERL5LIB:+:}${perlPackages.Po4a}/lib/perl5";
```

Comment:
I am not sure why here is ```lib/perl5``` and not ```lib/perl5/site_perl```
Perhaps it never worked and is not needed at all

Reply:
seems like a separated issue, some perl libs (po4a, libapparmor, ...) are installed in ```$out/lib/perl5``` which introduces some irregularity and problems with finding the libs (for example ```${perlPackages.Po4a}/bin/po4a``` does not work looking for the libs in ```${perlPackages.Po4a}/lib/perl5/site_perl``` while they are in ```${perlPackages.Po4a}/lib/perl5```

Reply:
https://github.com/NixOS/nixpkgs/pull/52137
I'll cherry-pick it here